.travis.yml

```diff
@@ -0,0 +1,44 @@
+language: android
+jdk: oraclejdk8
+sudo: required # See https://github.com/travis-ci/travis-ci/issues/5582
+
+android:
+  components:
+    - tools
+    - tools
+    - platform-tools
+    - android-26
+    - android-27
+    - build-tools-28.0.3
+    - sys-img-x86-google_apis_playstore-26
+  licenses:
+    - 'android-sdk-preview-license-52d11cd2'
+    - 'android-sdk-preview-license-d099d938'
+    - 'android-sdk-license-.+'
+    - 'google-gdk-license-.+'
+  before-install:
+    - echo yes | sdkmanager "platforms;android-26"
+    - echo yes | sdkmanager "platforms;android-27"
+    - echo yes | sdkmanager "platforms;android-28"
+    - echo yes | sdkmanager "build-tools;28.0.3"
+    - mkdir "$ANDROID_HOME/licenses" || true
+    - echo -e "\n8933bad161af4178b1185d1a37fbf41ea5269c55\n24333f8a63b6825ea9c5514f83c2829b004d1fee" > "$ANDROID_HOME/licenses/android-sdk-license"
+    - echo -e "\n84831b9409646a918e30573bab4c9c91346d8abd\n504667f4c0de7af1a06de9f4b1727b84351f2910" > "$ANDROID_HOME/licenses/android-sdk-preview-license"
+  
+  env:
+    global:
+      # install timeout in minutes (2 minutes by default)
+      - ADB_INSTALL_TIMEOUT=8
+
+  before_script:
+    - echo no | android create avd --force -n test -t android-26 --abi x86 -c 200M
+    - android list targets
+    - emulator -avd test -no-audio -no-window &
+    - android-wait-for-emulator
+    - adb shell input keyevent 82 &
+  
+  script:
+    - ./gradlew build 
+    - ./gradlew :library:test 
+    # Travis refuses to start the damn emulator
+    # - ./gradlew :app:connectedDebugAndroidTest
```

Comment:
If you ever figure out how to start the emulator on Travis please let me know. I've also given up